src/data-workspace/form-wrapper.js

```diff
@@ -21,29 +20,34 @@ function createInitialValues(dataValueSet) {
     ])
 }
 
-const onSubmit = (values, form) => console.log({ values, form })
-
-const subscriptions = {}
-
-export function FinalFormWrapper({ children, dataValueSet }) {
+export function FormWrapper({ children, dataValueSet, validFormKey }) {
     // This prevents rerendering on every data value update, but also prevents
     // reinitializing (and populating) an empty form when going back online.
     // TODO: Reinitialize form `onSuccess` of dataValueSets query
     // See https://dhis2.atlassian.net/browse/TECH-1357
     const [initialValues] = useState(() => createInitialValues(dataValueSet))
-    return (
-        <Form
-            onSubmit={onSubmit}
-            initialValues={initialValues}
-            subscriptions={subscriptions}
-            keepDirtyOnReinitialize
-            mutators={{ setFieldData }}
-        >
-            {() => children}
-        </Form>
+
+    const setInitialDataValues = useValueStore(
+        (state) => state.setInitialDataValues
     )
+
+    const initialValuesInStore =
```

Comment:
This is the logic to check that the values in the zustand store are for the given data form, so as to prevent rendering the input cells before the initial values are present

Reply:
I found a small (but scary) bug that I think is related to this, but also the service worker somehow.

If you edit a cell - blur and let it save - then refresh. The value will not be shown in the form. It seems like we're getting data from the service worker that is not updated.

Reply:
😱 thanks for that!

I'm too be honest a little confused with the react-query nomenclature here. I would have thought that since we had `refetchOnMount: false` for this query that `isFetchedAfterMount` would remain false, but I guess it's 'isFetchedAfterMount` and maybe I'm imaging a variable `isRefetchedAfterMount`?